Framework/PythonInterface/test/python/mantid/api/AnalysisDataServiceObserverTest.py

```diff
@@ -0,0 +1,248 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2017 ISIS Rutherford Appleton Laboratory UKRI,
+#     NScD Oak Ridge National Laboratory, European Spallation Source
+#     & Institut Laue - Langevin
+# SPDX - License - Identifier: GPL - 3.0 +
+#  This file is part of the mantidqt package
+#
+from __future__ import (absolute_import, division, print_function, unicode_literals)
+
+import unittest
+import sys
+
+from mantid.api import AnalysisDataService as ADS, AnalysisDataServiceObserver
+from mantid.simpleapi import CreateSampleWorkspace, RenameWorkspace, GroupWorkspaces, UnGroupWorkspace, DeleteWorkspace
+
+if sys.version_info.major == 3:
```

Comment:
I'd be tempted to leave it as 3 simply because it's not necessarily guaranteed that Python 4 will keep the same method of Importing Mock (Unless it's been announced somewhere and for which I apologize) and so would it not be better that it doesn't do the defined Python 3 version on everything greater than 3?

Reply:
It is highly likely that Python 4 will be Python 3 compatible.  It is very unlikely that Python 4 will revert to Python 2 behavior.

There is some interesting related discussion [here](http://astrofrog.github.io/blog/2016/01/12/stop-writing-python-4-incompatible-code/).

Reply:
Ahhh I appreciate the discussion link! It makes sense to do it that way given the compatibility of Python 4. (not promising anything on Python 5 but the code likely will have been replaced or disused by then anyways)